oak/server/dev/dev_oak_runner_main.cc

```diff
@@ -22,9 +22,38 @@
 #include "absl/synchronization/notification.h"
 #include "absl/time/time.h"
 #include "asylo/util/logging.h"
+#include "oak/common/utils.h"
+#include "oak/server/base_credential_provider.h"
 #include "oak/server/dev/dev_oak_loader.h"
+#include "oak/server/dev/insecure_credential_provider.h"
+#include "oak/server/dev/tls_credential_provider.h"
 
 ABSL_FLAG(std::string, application, "", "Application configuration file");
+ABSL_FLAG(std::string, credential_type, "insecure",
+          "The type of credentials to use: insecure or tls");
+ABSL_FLAG(std::string, ca_cert, "", "Path to the PEM-encoded CA root certificate");
+ABSL_FLAG(std::string, private_key, "", "Path to the private key");
+ABSL_FLAG(std::string, cert_chain, "", "Path to the PEM-encoded certificate chain");
+
+std::unique_ptr<oak::BaseCredentialProvider> GetCredentialProvider() {
+  switch (oak::utils::parse_credential_type(absl::GetFlag(FLAGS_credential_type))) {
+    case oak::utils::CredentialType::kInsecure: {
+      return absl::make_unique<oak::InsecureCredentialProvider>();
+    };
+    case oak::utils::CredentialType::kTls: {
+      auto ca_cert = oak::utils::read_file(absl::GetFlag(FLAGS_ca_cert));
+      auto private_key = oak::utils::read_file(absl::GetFlag(FLAGS_private_key));
+      auto cert_chain = oak::utils::read_file(absl::GetFlag(FLAGS_cert_chain));
```

Comment:
Maybe have some files-present/non-empty checks here to make it faster to spot a misconfigured server?

Reply:
Added checks that the file paths are specified. read_file will LOG(FATAL) if the files are not found. The GRPC server validates the certificate and key contents on startup.